src/metabase/models/setting.clj

```diff
@@ -1342,3 +1343,42 @@
            (map (fn [[setting-name]]
                   [setting-name (get setting-name)])))
      @registered-settings)))
+
+(defn- set-via-env? [setting]
+  (contains? env/env (setting-env-map-name setting)))
+
+(defmulti ^:private validate-setting-format :type)
+
+;; Treat setting as valid unless a validator has been defined for its type
+(defmethod validate-setting-format :default [_] nil)
+
+(defmethod validate-setting-format :json [{:keys [name]}]
+  (try
+    ;; attempt to parse the string to trigger any related exception
+    (doall (json/parse-string (get-value-of-type :string name)))
+    ;; there was no issue parsing the setting, so return a falsey value
+    nil
+    (catch JsonParseException e e)
```

Comment:
There's no test for this case.

Reply:
This is the case that's throwing in the current test, at least when I ran it in my REPL a few tweaks ago. Testing both cases is a fair point, but I'm not sure how best to trigger a non-JSON exception. If you have any ideas, let me know.